### PR TITLE
Wait for cleanup to be done after stop VMM

### PR DESCRIPTION
**Pull request by Joffref**
Closes: #428

*Description of changes:* add a cleanup chan to ensure cleanup is waited by the stopVMM method.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Sugate <mariusjoffre@gmail.com>

### CODE REVIEW

**Comment by ginglis13** [comment]
Hi @Joffref , thanks for the PR! Could you please add DCO to your commits by signing them `git commit -s`?

**Comment by Joffref** [comment]
Is this good @ginglis13 ?

**Review by ginglis13 — CHANGES_REQUESTED** [review]
small nitpick, otherwise LGTM

File: machine_test.go [hunk 1]
```diff
@@ -799,6 +800,69 @@ func testStopVMM(ctx context.Context, t *testing.T, m *Machine) {
...
+			Smt:         Bool(false),
+		},
+	}
+	ctx := context.Background()
+	cmd := VMCommandBuilder{}.
+		WithSocketPath(cfg.SocketPath).
+		WithBin(getFirecrackerBinaryPath()).
+		Build(ctx)
+	m, err := NewMachine(ctx, cfg, WithProcessRunner(cmd), WithLogger(fctesting.NewLogEntry(t)))
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (Joffref) [reply machine_test.go:859]: Change done ! I'm letting you close the comment :+1:

**Comment by ginglis13** [comment]
LGTM, thanks. 2 quick requests before we can merge this:

1/ Can you squash these 5 commits into one commit?

2/ Can you sign that commit using your full, real first and last name?

**Review by ginglis13 — APPROVED** [review]

**Review by ginglis13 — APPROVED** [review]
LGTM, GH actions tests failing having been fixed in main already.

**Review by austinvazquez — APPROVED** [review]

**Pull request merged**